node/primitives/src/lib.rs

```diff
@@ -287,6 +291,82 @@ pub struct AvailableData {
 	pub validation_data: PersistedValidationData,
 }
 
+/// This is a convenience type to allow the Erasure chunk proof to Decode into a nested BoundedVec
+#[derive(PartialEq, Eq, Clone, Debug, Hash)]
+pub struct Proof(BoundedVec<BoundedVec<u8, 1, MERKLE_NODE_MAX_SIZE>, 1, MERKLE_PROOF_MAX_DEPTH>);
+
+impl Proof {
+	/// This function allows to convert back to the standard nested Vec format
+	pub fn as_vec(&self) -> Vec<Vec<u8>> {
```

Comment:
nit/future: Do we truely require a `Vec<Vec<_>>`? Or could we avoid all those allocations by employing a single `BoundedVec<_>` and providing a `Vec<&[]>` or even `&[&[]]`? Again, not for this PR, a potential future optimization.